modules/oas-page-builder/src/services/pageBuilder.ts

```diff
@@ -15,21 +15,36 @@ const fetchTextData = async (url: string, errMsg: string) => {
   return res.text();
 };
 
-const getOASFileUrl = (gitHash: string) => `${OAS_FILE_SERVER}${gitHash}.json`;
+interface AtlasSpecUrlParams {
+  apiKeyword: string;
+  apiVersion?: string;
+  resourceVersion?: string;
+}
 
-const getAtlasSpecUrl = async (apiKeyword: string) => {
+const getAtlasSpecUrl = async ({ apiKeyword, apiVersion, resourceVersion }: AtlasSpecUrlParams) => {
   // Currently, the only expected API fetched programmatically is the Cloud Admin API,
   // but it's possible to have more in the future with varying processes.
   const keywords = ['cloud'];
   if (!keywords.includes(apiKeyword)) {
     throw new Error(`${apiKeyword} is not a supported API for building.`);
   }
 
+  let versionExtension = '';
+
+  if (apiVersion) {
+    versionExtension += `-${apiVersion.split('.')[0]}`;
```

Comment:
Thanks for stripping away the decimal! Seems like the `v` is still missing from the version extension though. Should be: `-v${apiVersion.split('.')[0]}`.

Reply:
oops!! will add that